sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/spark/source/GpuSparkBatchQueryScan.java

```diff
@@ -165,9 +167,10 @@ private static TableScan buildScan(Scan cpuInstance,
   }
 
   GpuSparkBatchQueryScan(SparkSession spark, Table table, TableScan scan, SparkReadConf readConf,
-                         Schema expectedSchema, List<Expression> filters, RapidsConf rapidsConf) {
+                         Schema expectedSchema, List<Expression> filters, RapidsConf rapidsConf,
+                         boolean queryUsesInputFile) {
```

Comment:
This helped me found a bug. Thanks.
In GpuParquetScan, it will not fall back to CPU when COALESCING is set and `queryUsesInputFile ` is true, instead MULTI-THREADED reading will be picked.
I updated the `multiFileReadCheck` function to follow the above behavior, and add a test to at least make sure the query with input file works correctly.

We have[ unit tests to check the correctness of picking the reader type for various cases](https://github.com/NVIDIA/spark-rapids/blob/branch-22.10/tests/src/test/scala/com/nvidia/spark/rapids/GpuReaderTypeSuites.scala). Maybe we also need this for Iceberg.